In natty, $GDM_LANG doesn't always contain a full valid locale. It might
contain a "short" one (e.g. "fr") that is expanded (e.g. to "fr_fr.UTF-8)
by an Xsession script shipped by gdm. So drop the compatibility code from
xfce4-session, otherwise the session locale might be set to an invalid
value.

--- xfce4-session-4.8.1.orig/xfce4-session/main.c
+++ xfce4-session-4.8.1/xfce4-session/main.c
@@ -101,14 +101,6 @@ setup_environment (void)
   /* pass correct DISPLAY to children, in case of --display in argv */
   g_setenv ("DISPLAY", gdk_display_get_name (gdk_display_get_default ()), TRUE);
 
-  /* this is for compatibility with the GNOME Display Manager */
-  lang = g_getenv ("GDM_LANG");
-  if (lang != NULL && strlen (lang) > 0)
-    {
-      g_setenv ("LANG", lang, TRUE);
-      g_unsetenv ("GDM_LANG");
-    }
-
   /* check access to $ICEAUTHORITY or $HOME/.ICEauthority if unset */
   if (g_getenv ("ICEAUTHORITY"))
     authfile = g_strdup (g_getenv ("ICEAUTHORITY"));
